### PR TITLE
GUM-35:  Added navigation types

### PR DESCRIPTION
**Pull request by katyakostikova**

### CODE REVIEW

File: src/common/types/navigation/params/app-stack-param.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ type AppStackParamList = {
     isTeamOkr: boolean;
   };
   [AppRoute.ADD_KEY_RESULT]: undefined;
+  [AppRoute.ONBOARDING_SETUP]: undefined;
```
Comment (R-Mielamud) [inline src/common/types/navigation/params/app-stack-param.ts:12]: better use `void`

File: src/common/types/navigation/params/onboarding-stack-param.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import { AppRoute, OnboardingRoute } from 'src/common/enums';
+import { OnboardingRoute } from 'src/common/enums';
 
 type OnboardingStackParamList = {
   [OnboardingRoute.ONBOARDING]: undefined;
   [OnboardingRoute.ADD_EDUCATION]: undefined;
   [OnboardingRoute.ADD_EXPERIENCE]: undefined;
   [OnboardingRoute.ADD_LANGUAGE]: undefined;
   [OnboardingRoute.ADD_LOCATION]: undefined;
-  [AppRoute.APP]: undefined;
```
Comment (R-Mielamud) [inline src/common/types/navigation/params/onboarding-stack-param.ts:9]: Also `void`

File: src/common/types/navigation/params/profile-stack-param.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { ProfileRoute } from 'src/common/enums';
+
+type ProfileStackParamList = {
+  [ProfileRoute.PROFILE_ROOT]: undefined;
+  [ProfileRoute.CREATE_SKILL]: undefined;
```
Comment (R-Mielamud) [inline src/common/types/navigation/params/profile-stack-param.ts:5]: Same here and check all other files.

File: src/common/types/navigation/props/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from './roor-navigation-props';
```
Comment (R-Mielamud) [inline src/common/types/navigation/props/index.ts]: Typo `rooR` -> `rooT`

File: src/common/types/navigation/params/app-stack-param.ts (hunk 1 above)
Comment (katyakostikova) [reply src/common/types/navigation/params/app-stack-param.ts:12]: Are you sure ? [https://reactnavigation.org/docs/typescript/#type-checking-the-navigator](https://reactnavigation.org/docs/typescript/#type-checking-the-navigator)

Comment (R-Mielamud) [reply src/common/types/navigation/params/app-stack-param.ts:12]:
```typescript
type void = null | undefined
```

Comment (AleksandrTermenzhy) [reply src/common/types/navigation/params/app-stack-param.ts:12]: Common practice here is to use `undefined`. We can leave it as is

**Review by liza-veis — APPROVED** [review]

**Review by R-Mielamud — APPROVED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Pull request merged**